### DISCUSSION TITLE
Remap and alternate function PB3

### DISCUSSION THREAD

**Discussion by babitsyn** [CommunityGD32Cores/GD32Core-New#81] [Q&A]
Cannot run blinky project at PB3 pin. At PC13 blinky project works perfectly because GPIO is default function of this pin.  But PB3 default function is JTDO. How to set (or remap) PB3 in OUTPUT_OPEN_DRAIN mode?

**Comment by maxgerhardt** [comment]
I assume that's on GD32F303CC?

**Reply by babitsyn** [reply]
No, it is GD32F303CG. But I set GD32F303CC Generic, because it almost same.

**Reply by maxgerhardt** [reply]
I see. Yes we're definitely missing the remap setting for the default JTAG pins in the core, I will correct it when I'm home. We're missing a call to gpio_pin_remap_config with the JTAG remapping as defined in https://github.com/CommunityGD32Cores/ArduinoCore-GD32/blob/main/system/GD32F30x_firmware/GD32F30x_standard_peripheral/Include/gd32f30x_gpio.h

**Reply by babitsyn** [reply]
Working code: 
` // Blinky mod`
`#include <Arduino.h>`
`#define LED PB3`
`static int i=0;`
`void setup(){`
`    rcu_periph_clock_enable(RCU_AF);`
`    gpio_pin_remap_config(GPIO_SWJ_SWDPENABLE_REMAP, ENABLE);  `
`    //gpio_init(GPIOB, GPIO_MODE_OUT_OD, GPIO_OSPEED_50MHZ, GPIO_PIN_3|GPIO_PIN_4);`
`    Serial.begin(115200);`
`    pinMode(LED, OUTPUT_OPEN_DRAIN);    `
`}`

`void loop(){`
`    digitalWrite(LED, LOW);`
`    //gpio_bit_set(GPIOB, GPIO_PIN_3|GPIO_PIN_4);`
`    delay(500);`
`    //gpio_bit_reset(GPIOB, GPIO_PIN_3|GPIO_PIN_4);`
`    digitalWrite(LED, HIGH);`
`    delay(500);`
`    Serial.println("Blinky nr. " + String(i++));`
`} // End `
PB4 added for test purposes, to be sure if PB3 is broken. Both of "digitalWrite" and "gpio_bit_(re)set" working well. 
**rcu_periph_clock_enable(RCU_AF);** is mandatory!
